src/F1_plotting.py

```diff
@@ -403,6 +413,112 @@ def plot_a_piechart(dict_values, settings, file_name, costs, label, title):
     return
 
 
+def convert_plot_data_to_dataframe(plot_data_dict, data_type):
+    """
+
+    Parameters
+    ----------
+    plot_data_dict: dict,
+        timeseries for either demand or supply
+    data_type: str,
+        one of "demand" or "supply"
+
+    Returns
+    -------
+    df: pandas:`pandas.DataFrame<frame>`,
+        timeseries for plotting
+    """
+    # Later, this dataframe can be passed to a function directly make the graphs with Plotly
+    df = pd.DataFrame.from_dict(plot_data_dict[data_type], orient="columns")
+
+    # Change the index of the dataframe
+    df.reset_index(level=0, inplace=True)
+    # Rename the timestamp column from 'index' to 'timestamp'
+    df = df.rename(columns={"index": "timestamp"})
+    return df
+
+
+def extract_plot_data_and_title(dict_values, df_dem=None):
+    """Dataframe used for the plots in the report
```

Comment:
I am not sure if this is a generic function or only appliccable to plot energyConsumption and energyProduction timeseries. If possible, the docstring could be extended.

Reply:
Changed the docstring in 45987d772493143ea748db01e62f2a020c559094